addons/account/static/src/js/account_followup_report_widgets.js

```diff
@@ -0,0 +1,160 @@
+        var target_id = $("#nextActionModal #target_id").val();
+        var date = $("#nextActionDate").val();
+        var contextModel = new openerp.Model('account.report.context.followup');
+        contextModel.call('change_next_action', [[parseInt(target_id)], date, note]).then(function (result) {
+            $('#nextActionModal').modal('hide');
+            date = new Date(date.substr(0, 4), date.substr(5, 2), date.substr(8, 2), 12, 0, 0, 0).toLocaleDateString();
+            $('div.page.' + target_id).find('.oe-account-next-action').html(openerp.qweb.render("nextActionDate", {'note': note, 'date': date}));
+        });
+    },
+    setNextAction: function(e) {
+        e.stopPropagation();
+        e.preventDefault();
+        var target_id = $(e.target).parents("div.page").attr("class").split(/\s+/)[3];
+        var dt = new Date();
+        switch($(e.target).attr("class").split(/\s+/)[2]) {
+            case 'oe-account-followup-one-week':
+                dt.setDate(dt.getDate() + 7);
+                break;
+            case 'oe-account-followup-two-weeks':
+                dt.setDate(dt.getDate() + 14);
+                break;
+            case 'oe-account-followup-one-month':
+                dt.setMonth(dt.getMonth() + 1);
+                break;
+            case 'oe-account-followup-two-months':
+                dt.setMonth(dt.getMonth() + 2);
+                break;
+        }
+        $("#nextActionModal #nextActionDate").val(dt.toISOString().substr(0, 10));
+        $("#nextActionModal #target_id").val(target_id);
+        $('#nextActionModal').on('hidden.bs.modal', function (e) {
+            $(this).find('form')[0].reset();
+        });
+        $('#nextActionModal').modal('show');
+    },
+    onChangeBlocked: function(e) {
+        e.stopPropagation();
+        e.preventDefault();
+        var checkbox = $(e.target).is(":checked")
+        var target_id = $(e.target).parents('tr').attr("class").split(/\s+/)[1];
+        var model = new openerp.Model('account.move.line');
+        model.call('write', [[parseInt(target_id)], {'blocked': checkbox}])
+    },
+    onKeyPress: function(e) {
+        var report_name = $("div.page").attr("class").split(/\s+/)[2];
+        if ((e.which === 13 || e.which === 10) && (e.ctrlKey || e.metaKey) && report_name == 'followup_report') {
+            $('a.btn-primary.followup-email').trigger('click');
+            var letter_context_list = [];
+            $('a.btn-primary.followup-letter').each(function() {
+                letter_context_list.push($(this).attr('context'))
+            });
+            window.open('?pdf&letter_context_list=' + letter_context_list, '_blank');
+            window.open('?partner_done=all', '_self');
+        }
+    },
+    skipPartner: function(e) {
+        var partner_id = $(e.target).attr("partner");
+        var model = new openerp.Model('res.partner');
+        if ($(e.target).attr('class') == 'btn btn-primary followup-skip') {
```

Comment:
only one special class instead of several and do not base yourself on btn-primary since it can be change for design purpose